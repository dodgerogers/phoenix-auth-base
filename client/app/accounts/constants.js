import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'GET_CURRENT_USER_REQUEST',
  'GET_CURRENT_USER_SUCCESS',
  'GET_CURRENT_USER_FAILURE',
  'GET_CURRENT_USER_PROFILES_REQUEST',
  'GET_CURRENT_USER_PROFILES_SUCCESS',
  'GET_CURRENT_USER_PROFILES_FAILURE',
  'SET_CURRENT_USER_PROFILE',
  'GET_CURRENT_ACCOUNT_REQUEST',
  'REMOVE_CURRENT_ACCOUNT_REQUEST',
]);
