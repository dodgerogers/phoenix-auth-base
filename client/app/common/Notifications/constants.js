import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'NOTIFY_SUCCESS',
  'NOTIFY_FAILURE',
  'DESTROY',
]);

export const areaIDs = mirrorCreator([
  'APPLICATION',
  'AUTHENTICATION',
]);
