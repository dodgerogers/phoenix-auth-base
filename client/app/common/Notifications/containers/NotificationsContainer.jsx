import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import NotificationList from '../components/NotificationList';
import { destroy } from '../actions';

const mapStateToProps = (state, ownProps) => {
  return {
    notifications: state.notifications.get(ownProps.id),
  };
};

export default connect(mapStateToProps, { destroy })(NotificationList);
