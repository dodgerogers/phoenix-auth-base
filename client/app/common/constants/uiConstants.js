export const colors = {
  RedRibbon: '#E90F3A',
  Carmine: '#a70122',
  RedViolet: '#ea1aa3',
  HaitiPurple: '#120d36',
  ElectricViolet: '#a923e9',
  WindsorPurple: '#580c79',
  ParisMPurple: '#270864',
  PurpleHeart: '#612cc8',
  JacksonsPurple: '#1f28a5',
  CuriousBlue: '#2a8cbd',
  Turquoise: '#2fd6c2',
  Elm: '#1b9487',
  Shamrock: '#2ed583',
  GenoaGreen: '#1a985a',
  Apple: '#2dd535',
  BilbaoGreen: '#138119',
  AtlantisGreen: '#b4d535',
  Olivetone: '#627516',
  HokeyPokeyGold: '#d6aa32',
  ReefGold: '#9e791b',
  Tango: '#ef7d25',
  HawaiianTan: '#a05315',
  Pomegranate: '#ee3b1f',
  Tabasco: '#a72913',
  Sangria: '#8f030a',
  DullLavender: '#9b89e6',
  Crimson: '#e61740',
};
