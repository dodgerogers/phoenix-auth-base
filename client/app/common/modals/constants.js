import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'INITIALIZE_MODAL',
  'SHOW_MODAL',
  'HIDE_MODAL',
]);

export const modalIds = mirrorCreator([
  'LOGIN_MODAL',
  'REGISTRATION_MODAL',
  'CONFIRMATION_MODAL',
  'RESEND_CONFIRMATION_MODAL',
  'FORGOT_PASSWORD_MODAL',
  'RESET_PASSWORD_MODAL',
])
