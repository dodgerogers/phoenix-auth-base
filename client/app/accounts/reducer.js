import { fromJS } from 'immutable';
import { actionTypes } from './constants';

export const initialState = fromJS({
  currentUser: null,
  currentProfile: null,
  userProfiles: [],
});

export default function accountsReducer(state = initialState, action) {
  const { type } = action;

  switch (type) {
    case actionTypes.GET_CURRENT_USER_SUCCESS:
      return state.set('currentUser', fromJS(action.user));
    case actionTypes.GET_CURRENT_USER_PROFILES_SUCCESS:
      return state.set('userProfiles', fromJS(action.profiles));
    case actionTypes.SET_CURRENT_USER_PROFILE:
      const profile = state.userProfiles.find(p => p.id === action.profileId);
      return state.set('currentProfile', profile);
    case actionTypes.GET_CURRENT_USER_FAILURE:
      return state.set('currentUser', initialState.get('currentUser'));
    case actionTypes.GET_CURRENT_USER_PROFILES_FAILURE:
      return state.set('userProfiles', initialState.get('userProfiles'));
    default:
      return state;
  };
}
