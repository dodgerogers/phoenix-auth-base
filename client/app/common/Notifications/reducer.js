import { fromJS } from 'immutable';
import { actionTypes, areaIDs } from './constants';
import uid from '../../lib/utils/UidGenerator';


const initialNotification = fromJS({
  level: null,
  message: null,
});

const initialList = fromJS([]);

export const initialState = fromJS({
  [areaIDs.APPLICATION]: initialList,
  [areaIDs.AUTHENTICATION]: initialList,
});

export default function notificationReducer(state = initialState, action) {
  const { type } = action;

  switch (type) {
    case actionTypes.NOTIFY:
      return state.update(getAreaID(action), array => array.push(incrementID(action.notification, array)));
    case actionTypes.DESTROY:
      return state.update(getAreaID(action), list => filterById(list, action.notification));
    default:
      return state;
  };
}

const getAreaID = (action) => {
  return action.id || areaIDs.APPLICATION;
}

const incrementID = (notification, array) => {
  const lastNotification = array.get(-1);
  const id = lastNotification ? lastNotification.get('id') + 1 : 0;
  notification.id = id;

  return fromJS(notification);
}

const filterById = (list, notification) => {
  return list.filter(n => n.get('id') !== notification.get('id'))
}
