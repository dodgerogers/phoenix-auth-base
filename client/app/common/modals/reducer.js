import { fromJS, Map } from 'immutable';
import { actionTypes } from './constants';

export const initialState = fromJS({});

export default function modalsReducer(state = initialState, action) {
  const { type } = action;

  switch (type) {
    case actionTypes.SHOW_MODAL:
      return showModal(state, action.data.id);
    case actionTypes.HIDE_MODAL:
      return hideModal(state, action.data.id);
    default:
      return state;
  }
}

function showModal(state, id) {
  return state.merge(hideAllModals(state)).set(id, true);
}

function hideModal(state, id) {
  return state.merge(hideAllModals(state)).set(id, false);
}

function hideAllModals(state) {
  return fromJS(state.keySeq().reduce((memo, key) => {
    memo[key] = false;
    return memo;
  }, {}));
}
