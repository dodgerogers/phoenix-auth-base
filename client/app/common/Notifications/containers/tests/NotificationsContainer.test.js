import React from 'react';
import { fromJS } from 'immutable';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState } from '../../reducer';
import { areaIDs } from '../../constants';
import NotificationsContainer from '../NotificationsContainer';

const mockStore = configureStore([thunk]);

describe('NotificationsContainer', () => {
  const fullProps = () => ({
    id: areaIDs.APPLICATION,
  });

  describe('connected component', () => {
    const connectedComponent = (store, props) => mount(
      <Provider store={store}>
        <NotificationsContainer {...props} />
      </Provider>
    );

    it('matches snapshot', () => {
      const store = mockStore({ notifications: initialState });
      const props = fullProps();
      const component = connectedComponent(store, props);

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
