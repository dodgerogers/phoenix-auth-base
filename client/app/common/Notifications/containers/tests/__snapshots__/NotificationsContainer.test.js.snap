// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsContainer connected component matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(NotificationList)
    id="APPLICATION"
  >
    <NotificationList
      destroy={[Function]}
      id="APPLICATION"
      notifications={
        Immutable.List [
        ]
      }
    >
      <noscript />
    </NotificationList>
  </Connect(NotificationList)>
</Provider>
`;
